base_phone_popup/models/phone_common.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2014-2016 Akretion, Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+class PhoneCommon(models.AbstractModel):
+    _inherit = 'phone.validation.mixin'
+
+    @api.model
+    def get_record_from_phone_number(self, presented_number):
+        """If it finds something, it returns (object name, ID, record name)
+        For example : ('res.partner', 42, u'Alexis de Lattre (Akretion)')
+        """
+        _logger.debug(
+            "Call get_name_from_phone_number with number = %s"
+            % presented_number)
+        if not presented_number.isdigit():
+            _logger.warning(
+                "Number '%s' should only contain digits." % presented_number)
+        nr_digits_to_match_from_end = \
+            self.env.user.company_id.number_of_digits_to_match_from_end
+        if len(presented_number) >= nr_digits_to_match_from_end:
+            end_number_to_match = presented_number[
```

Comment:
I think the logic should be the contrary: you want to strip extra numbers at the end, not at the beginning. Any way, as it is, you don't need to make the `if`: `presented_number[-nr_digits_to_match_from_end:]` always return correct thing. But as said, I doubt this is the best approach. @alexis-via what is the reason for stripping initial numbers?

Reply:
Reading the field help, I would say that the technique should be to normalize the inbound number before comparing. What I said about stripping final numbers is because some switchboard add the extension number at the end, and this is something that we can handle searching first for full match, and if not, try to strip that part.

Reply:
The reason to strip initial number is the following: from my experience, calls from abroad are regularly badly presented by some telco without the country prefix, etc... So matching on the end of a phone number avoid this kind of trouble on incoming calls with foreign numbers without proper country prefix.

Reply:
OK, we can put both options.